### PR TITLE
Brainzplayer bug fixes

### PR DESCRIPTION
**Pull request by yellowHatpro**
- bug fixed: The inability to pause play song again.
- playlist switching to be enabled
     1. Currently made a class Playable that can take songs of songs, selected albums, artists, and playlists.
     2. This class will be communicating with the BrainzPlayer service and providing songs to MediaSource.

### CODE REVIEW

**Comment by 07jasjeet** [comment]
Some issues I've been encountering : 
1) Music being autoplayed when entering the app.
2) Not being able to close the notification player even though no music is playing.
3) The animations of mini player.
4) Viewpager's (miniplayer) background persists even though no music is found.

Note : I made modifications to DISC_NUMBER variable with an alternative because the app was crashing as the disc number variable required min sdk 30. Do check if its right or wrong (It works for the devices i've tested).

I can help you if you want after i'm finished with yim.

**Comment by yellowHatpro** [comment]
@07jasjeet thanks for informing . I have addressed 1st, and 3rd issues and will be making PR soon. 
Regarding 2nd, it is because of Media Controls, and they have to be removed manually. (https://developer.android.com/guide/topics/media/media-controls)
Regarding the 4th, I will try to come up with a fix. Should we completely remove the bottom viewpager in case of no song?

**Comment by yellowHatpro** [comment]
Also for DISC_NUMBER, I think @akshaaatt has made changes.
MediaStore.Audio.AudioColumns.COMPOSER as a replacement for DISC_NUMBER

**Comment by 07jasjeet** [comment]
I suggested the change and he crosschecked it but i thought I'd be best that you were informed as you are the creator of bp. Crazy how big bp is! Respect++

**Comment by yellowHatpro** [comment]
Thanks 😁!!
Alright, then I'll remove the song viewpager in that case.
Yupp I have created a new class for that purpose, so that the song and its respective playable parent (Album, Artist, or playlist) is saved.

**Comment by 07jasjeet** [comment]
Thankyou, glad you liked it!!

File: app/src/main/java/org/listenbrainz/android/data/sources/brainzplayer/Playable.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.listenbrainz.android.data.sources.brainzplayer
+
+data class Playable(
+    val type: String,
+    val id: Long,
+    val songs: List<Song>,
+    val currentlyPlayingSongIndex: Int
```
Comment (akshaaatt) [inline app/src/main/java/org/listenbrainz/android/data/sources/brainzplayer/Playable.kt:7]: Change this to `currentSongIndex` I suppose

**Review by akshaaatt — COMMENTED** [review]
A few changes then we should be good to go!

File: app/src/main/java/org/listenbrainz/android/util/SharedPrefManager.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package org.listenbrainz.android.util
+
+import android.content.Context
+import android.content.SharedPreferences
+import org.listenbrainz.android.BuildConfig
+import org.listenbrainz.android.data.di.brainzplayer.TypeConverter
+import org.listenbrainz.android.data.sources.brainzplayer.Playable
+
+class SharedPrefManager(val context: Context) {
```
[the remark this answers is not in the document]
Comment (yellowHatpro) [reply app/src/main/java/org/listenbrainz/android/util/SharedPrefManager.kt:9]: Should I just rename the LoginSharedPreferences to some generic name?

[the remark this answers is not in the document]
Comment (yellowHatpro) [reply app/src/main/java/org/listenbrainz/android/util/SharedPrefManager.kt]: Done! verified, nothing broke

File: app/src/main/java/org/listenbrainz/android/data/sources/brainzplayer/Playable.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.listenbrainz.android.data.sources.brainzplayer
+
+data class Playable(
+    val type: String,
```
[the remark this answers is not in the document]
Comment (yellowHatpro) [reply app/src/main/java/org/listenbrainz/android/data/sources/brainzplayer/Playable.kt]: Done!

File: app/src/main/java/org/listenbrainz/android/data/sources/brainzplayer/Playable.kt (hunk 1 above)
Comment (yellowHatpro) [reply app/src/main/java/org/listenbrainz/android/data/sources/brainzplayer/Playable.kt]: Done!

**Review by akshaaatt — APPROVED** [review]
LGTM! Thank you! 💯

**Pull request merged**